src/processor.rs

```diff
@@ -52,28 +58,38 @@ impl ProcessorGeneration {
         let model = (ext_model << 4) | base_model;
         let family = base_family + ext_family;
 
-        let id = (model, family);
-
-        let milan = (1, 25);
-        let genoa = (17, 25);
-
-        if id == milan {
-            return Ok(Self::Milan);
-        } else if id == genoa {
-            return Ok(Self::Genoa);
+        match family {
+            0x19 => match model {
+                1..=15 => Ok(Self::Milan),
+                16..=31 => Ok(Self::Genoa),
+                160..=175 => match socket {
```

Comment:
We have a `match` statement with decimal and hex options. Should we settle for one universal representation?

```
0x1..0xF => Ok(Self::Milan),
...
```

Reply:
Good point. Fixed!